damnit/ctxsupport/ctxrunner.py

```diff
@@ -611,15 +611,26 @@ def save_hdf5(self, hdf5_path, reduced_only=False):
                     reduced_ds.attrs['summary_method'] = var_obj.summary or ''
 
         for name, obj in xarray_dsets:
-            # HDF5 doesn't allow slashes in names :(
-            if isinstance(obj, xr.DataArray) and obj.name is not None and "/" in obj.name:
-                obj.name = obj.name.replace("/", "_")
+            if isinstance(obj, xr.DataArray):
+                # HDF5 doesn't allow slashes in names :(
+                if obj.name is not None and "/" in obj.name:
+                    obj.name = obj.name.replace("/", "_")
+                obj.encoding.update(COMPRESSION_OPTS)
             elif isinstance(obj, xr.Dataset):
-                names = {name: name.replace("/", "_") for name in obj.data_vars
-                         if name is not None and "/" in name}
-                obj = obj.rename_vars(names)
-
-            obj.to_netcdf(hdf5_path, mode="a", format="NETCDF4", group=name, engine="h5netcdf")
+                vars_names = {}
+                for var_name, dataarray in obj.items():
+                    if var_name is not None and "/" in var_name:
+                        vars_names[var_name] = var_name.replace("/", "_")
+                    dataarray.encoding.update(COMPRESSION_OPTS)
+                obj = obj.rename_vars(vars_names)
+
+            obj.to_netcdf(
```

Comment:
[`to_netcdf()`](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.to_netcdf.html#xarray.DataArray.to_netcdf) takes an `encoding=` parameter. Does this work? If it does, it seems neater than modifying the objects to change how they're stored.

Reply:
if *should* work but you still need to map variable names to encoding args, i.e. `{'var1': {...}, 'var2': {...}, ...}`.  
Another caveat I've read is that it overwrites any potential existing information the dataarray encoding.

Reply:
OK, fair enough. Though that does make me think, should we apply the changes in a way that leaves existing values if they're there? Or does xarray create the array with default options?

Reply:
do you mean, e.g. not overwrite the compression algorithm or level if it is already defined?  
I think I'm more enclined to overwrite these always (at least the compression algorithm) 🤔

Reply:
I believe `encoding` is an empty dict by default, but I imagine we could use it in the `Variable` code do improve compression on a variable basis with chunking, quantization, ...

Reply:
> do you mean, e.g. not overwrite the compression algorithm or level if it is already defined?

That was what I was thinking, e.g. if you knew data was likely to be incompressible, you could tell it not to try by setting `arr.encoding['compression'] = None`. I don't feel strongly about this, though, so if you're happy overwriting them, go for it.

Reply:
I'd be fine with that, just worried (although I sense this is unlikely to be an issue) that someone tries to use a non standard compression filter, or one that isn't supported in our environment.

Reply:
Your call - I'm also happy to let it try compressing everything the same way and add more options later if the need arises. :slightly_smiling_face:

Reply:
I changed my mind and keep original encoding options from the data array.